woocommerce/woocommerce
# [Accessibility] Announce to screen readers when the variations have been cleared

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #43611
Closes #43609

- The `reset_variations` has been updated from a link to a button
- The `reset_variations` button adds cleared screen reader text
- The `reset_variations` button also updates a new screen reader `alert`, announcing to users when they activate it, 'Your selection has been reset. Please select some product options before adding this product to your cart.' (translated into the appropriate language)

**Note**: since the `reset_variations` button has changed from a link to a button, it does change the appearance of the button for the Storefront Theme



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Activate your screen reader of chose
2. Select a Product that includes variations
3. Tab to a variation, use your arrows and space bar to make a selection
4. repeat for all variations
5. tab to the "clear" button
6. press spacebar to activate the clear button
7. you should hear that 'Your selection has been reset. Please select some product options before adding this product to your cart.' (translated into the appropriate language)



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
